### PR TITLE
Goreleaser/Homebrew: Install shell completions

### PR DESCRIPTION
**Pull request by craSH**
### What does this PR do?

* Enhancement! This will automatically install shell completions, generated by the installed `stratus` binary in their proper locations for users of the Homebrew tap. This method from Homebrew supports bash, fish, and zsh. _Sorry PowerShellers._

### Motivation

I just wanted to have the formula install completions automatically in the most correct manner possible.

### Other

* [Homebrew formula documentation](https://rubydoc.brew.sh/Formula#generate_completions_from_executable-instance_method) for `generate_completions_from_executable` method
* [Goreleaser brew documentation](https://goreleaser.com/customization/homebrew/) for custom install definition

I've never actually used Goreleaser before, but I think I got this right! I tested as much as I could locally, but I have *not* actually tested the creation of the `Formula/stratus-red-team.rb.rb` file as that requires an actual deploy to a Git repo, and I didn't do so during my own testing.

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -22,6 +22,11 @@ brews:
   url_template: "https://github.com/DataDog/stratus-red-team/releases/download/{{ .Tag }}/{{ .ArtifactName }}"
   license: Apache-2.0
   homepage: "https://stratus-red-team.cloud"
+  install: |
+    bin.install "stratus"
+
+    # Install shell completions
+    generate_completions_from_executable(bin/"stratus", "completions", shells: [:bash, :fish, :zsh])
```
Comment (christophetd) [inline .goreleaser.yaml:29]:
```suggestion
    generate_completions_from_executable(bin/"stratus", "completion", shells: [:bash, :fish, :zsh])
```

c.f. 

```
$ stratus --help
completion  Generate the autocompletion script for the specified shell
```

File: .goreleaser.yaml [hunk 2]
```diff
@@ -22,6 +22,11 @@ brews:
   url_template: "https://github.com/DataDog/stratus-red-team/releases/download/{{ .Tag }}/{{ .ArtifactName }}"
   license: Apache-2.0
   homepage: "https://stratus-red-team.cloud"
+  install: |
+    bin.install "stratus"
+
+    # Install shell completions
+    generate_completions_from_executable(bin/"stratus", "completion", shells: [:bash, :fish, :zsh])
```
Comment (christophetd) [inline .goreleaser.yaml:29]:
```suggestion
    generate_completions_from_executable(bin/"stratus", "completion", shells: [:bash, :fish, :zsh], base_name: "stratus")
```

**Comment by christophetd** [comment]
Thanks for the contribution! I tested this locally after making minor changes, by adding the lines to the local `Formula/stratus-red-team.rb` file, uninstalling stratus red team and installing it from the local formula:

```
brew uninstall stratus-red-team
brew install --build-from-source ./Formula/stratus-red-team.rb
```

It seems to install the completions:

```
==> Reinstalling stratus-red-team
==> Caveats
zsh completions have been installed to:
  /usr/local/share/zsh/site-functions
```

But then in a new shell, if I do:

```
stratus d[tab]
```

It prints:

```
(eval):1: _stratus-red-team: function definition file not found
(eval):1: _stratus-red-team: function definition file not found
(eval):1: _stratus-red-team: function definition file not found
```

That seems to be because it expects `/usr/local/share/zsh/site-functions/_stratus-red-team` - but if I don't change the base name, `stratus d[tab]` doesn't detect autocompletion. Any idea how to solve it?

**Comment by craSH** [comment]
Thanks for the fixes! You're right that setting `base_name` as you did in [67e2136](https://github.com/DataDog/stratus-red-team/pull/352/commits/67e213673445a50cdaa920e50fa3556729a57ed0) should be the right way to install the completion scripts (I didn't catch that it uses the Formula name vs. the binary name by default).

I pulled your changes, ran the command you provided (I had to doctor up the version and hash to make it work with a published binary on github) and it all looks correct and works:

- bash completions installed: 
  `/opt/homebrew/etc/bash_completion.d/stratus@ ⇒ ../../Cellar/stratus-red-team/2.5.3/etc/bash_completion.d/stratus`
- fish completions installed:
  `/opt/homebrew/share/fish/vendor_completions.d/stratus.fish@ ⇒ ../../../Cellar/stratus-red-team/2.5.3/share/fish/vendor_completions.d/stratus.fish`
- zsh completion installed:
`/opt/homebrew/share/zsh/site-functions/_stratus@ ⇒ ../../../Cellar/stratus-red-team/2.5.3/share/zsh/site-functions/_stratus`

Tested zsh completions by reloading my shell:

```
$ stratus detonate[tab]
aws.credential-access.ec2-get-password-data                 aws.exfiltration.ec2-security-group-open-port-22-ingress    azure.execution.vm-run-command
aws.credential-access.ec2-steal-instance-credentials        aws.exfiltration.ec2-share-ami                              azure.exfiltration.disk-export
[...]
aws.execution.ec2-user-data
```

**Comment by christophetd** [comment]
Actually that seems to work as well, I may have forgotten to open a new shell after installing! Will add some docs and get it merged. Thanks for the great contribution!

**Comment by christophetd** [comment]
Now released as part of v2.5.4

**Pull request merged**